### ISSUE TITLE
[SOLVED] Crash related to not being logged into YouTube Live Control Panel Dock, when launching a streaming from fresh OS startup

### ISSUE THREAD

**Issue by RaienryuuNoNatsu** [obsproject/obs-studio#10420]
### Operating System Info

Windows 10

### OBS Studio Version

30.1.0

### OBS Studio Log URL

https://obsproject.com/logs/wBOCEy6mZ3SeSleK

### OBS Studio Crash Log URL

https://obsproject.com/logs/AAwMl7Pp5BlKVEfG

### Expected Behavior

Like the previous stable release, it shouldn't crash, as I haven't changed anything at all.

### Current Behavior

It crashes on first launch when starting the system up.

### Steps to Reproduce

1. Fresh boot
2. Launch stream (specifically YouTube)
3. It just crashes

### Anything else we should know?

Worth noting that, at least for it to work, I don't need to enter Safe Mode. I don't even use plugins or add-ons/widgets. On second launch, it works just fine. However, something I've been noticing is that it consistently disconnects and reconnects about 2 hours into the live.

This is the log after today's streaming. However, it doesn't crash, so this is all I have in this regard.
[2024-03-22 14-55-51.txt](https://github.com/obsproject/obs-studio/files/14727540/2024-03-22.14-55-51.txt)

**Comment by RytoEX** [comment]
I have so far been unable to reproduce this. It seems to be crashing somewhere here:
https://github.com/obsproject/obs-studio/blob/30.1.0/UI/window-dock-youtube-app.cpp#L246-L295

**Comment by RytoEX** [comment]
Some additional questions, just to confirm:
1. Did you use a stream key for the Stream Settings? Or did you connect your YouTube Account?
2. Did you set the Stream Settings up from scratch, or did you import/duplicate an existing profile?

**Comment by RaienryuuNoNatsu** [comment]
1. I have my YouTube account connected.
2. As of using 30.1.0, it was an existing profile. This specific one is a 1440p@30fps profile, but the same resolution at 60fps crashes like that. To further explain, I made a 1080p profile ages ago, before the major UI and features update, and just made duplicates off of that one, and have been using those ever since.

**Comment by RytoEX** [comment]
> 1. I have my YouTube account connected.
> 2. As of using 30.1.0, it was an existing profile. This specific one is a 1440p@30fps profile, but the same resolution at 60fps crashes like that. To further explain, I made a 1080p profile ages ago, before the major UI and features update, and just made duplicates off of that one, and have been using those ever since.

Did you sign in after the initial startup? The log indicates that you were not signed in.
> 14:55:53.729: YT: InitYTUserUrl() User is not signed
> 14:55:53.730: YT: InitYTUserUrl() User is not signed

There are some known quirks about duplicating profiles that are using service integration account connections (such as #7140). My main questions about this profile are based on: has this duplicate profile (the new one, not the base one) been used before in _this_ installation of OBS?

**Comment by RaienryuuNoNatsu** [comment]
Sorry, I was double checking my streaming archives and logs to make sure of when I began using this version and these profiles on this installation.

Oh, interesting. That's something I've never even noticed. I've always just made sure I was logged in under the Settings > Stream tab, which I'm always logged into. As far as the YouTube Dock Panel goes, I don't use it at all, I just have the classic chat log opened on the right. In regards to the YouTube Dock itself, I do have to log in, but I actually never did even in 30.0.0, and never had any issue whatsoever even ignoring it. This is as far as I can tell in terms of not being logged in.

Like, this is my general and simple setup.
![immagine](https://github.com/obsproject/obs-studio/assets/31509027/614eeb6b-5e08-4334-8031-091e561506a4)

But, if I enable the Dock, it turns out I'm never logged in. Now, does it log me in when I launch stream? Maybe, but as I mentioned, I don't use it.
![immagine](https://github.com/obsproject/obs-studio/assets/31509027/a267ec6b-57da-4ac1-a2e5-73c13affd644)

About the second question: So, I have used the 1440p@30fps (the duplicate) one on the 18th on this installation of OBS. On that day, it didn't crash. For reference, the first time I used the 1440p@60fps profile (another duplicate), it crashed straight away. And today was the former 30fps one crashing. Also, the streaming settings for these profiles are the same if not for framerate and bitrate caps.

**Comment by RytoEX** [comment]
> In regards to the YouTube Dock itself, I do have to log in, but I actually never did even in 30.0.0, and never had any issue whatsoever even ignoring it. This is as far as I can tell in terms of not being logged in.

That's weird, since the crash is in the functions related to the dock that you are not logged into.

> But, if I enable the Dock, it turns out I'm never logged in. Now, does it log me in when I launch stream? Maybe, but as I mentioned, I don't use it.

That's expected. The YouTube Live Control Room dock requires its own separate login. It does not share an authenticated session with OBS.

> About the second question: So, I have used the 1440p@30fps (the duplicate) one on the 18th on this installation of OBS. On that day, it didn't crash. For reference, the first time I used the 1440p@60fps profile (another duplicate), it crashed straight away.

This makes me think there is some value expected to be set that is not after duplicating a profile.

**Comment by RytoEX** [comment]
I do notice that you're running in Portable Mode. Did you import these profiles, or did you duplicate them in-place while in Portable Mode?

**Comment by RaienryuuNoNatsu** [comment]
I duplicated them in-place. The deal is, I've been running this setup for quite some time now. Each profile is a duplicate in itself. As an example, I might've done my first stream at 1080p@60fps as my original profile back in version 26 or something, and over time and versions, I just duplicated that to make a 1440p profile, which then I used to create the framerate specific ones. At the time I wasn't even using Portable Mode, I switched alongside the new UI update, so I don't remember specifically if I was already in Portable mode when making this one specifically. For sure, however, I did everything always in-place.

**Comment by RytoEX** [comment]
> I duplicated them in-place. The deal is, I've been running this setup for quite some time now. Each profile is a duplicate in itself. As an example, I might've done my first stream at 1080p@60fps as my original profile back in version 26 or something, and over time and versions, I just duplicated that to make a 1440p profile, which then I used to create the framerate specific ones. At the time I wasn't even using Portable Mode, I switched alongside the new UI update, so I don't remember specifically if I was already in Portable mode when making this one specifically. For sure, however, I did everything always in-place.

I am not criticizing your chosen process. I'm just trying to boil down reliable steps to reproduce the crash, because I am still unable to do so. There must be some very specific set of steps needed to get there.

**Comment by RaienryuuNoNatsu** [comment]
Oh, I didn't take it the wrong way. I just tried to go back a bit so to make sure.

I wanna throw a detail in there that could be unrelated, but maybe not. So, ever since running in Portable Mode, I noticed that the Updater doesn't work. I like how I can just take my installation anywhere and just mirror my assets to I don't have to remake scenes everytime, and it doesn't bother me to come over to Github and grab the archive manually.
However, I simply drag, drop, and replace files. Any chances that there could be leftovers of 30.0.2 (last version I had) that are conflicting with 30.1.0? So maybe completely delete the 3 main folders and go fresh?

**Comment by RaienryuuNoNatsu** [comment]
Just in case, I went ahead and deleted the 3 main folders in the zip, and redownloaded v30.1.0. I expected it to not crash, as it never does after the first time, but I wanted to upload this log, in case something pops up that is different from the earlier ones.
[2024-03-23 01-58-04.txt](https://github.com/obsproject/obs-studio/files/14729690/2024-03-23.01-58-04.txt)

**Comment by RytoEX** [comment]
> Oh, I didn't take it the wrong way. I just tried to go back a bit so to make sure.
> 
> I wanna throw a detail in there that could be unrelated, but maybe not. So, ever since running in Portable Mode, I noticed that the Updater doesn't work. I like how I can just take my installation anywhere and just mirror my assets to I don't have to remake scenes everytime, and it doesn't bother me to come over to Github and grab the archive manually. However, I simply drag, drop, and replace files. Any chances that there could be leftovers of 30.0.2 (last version I had) that are conflicting with 30.1.0? So maybe completely delete the 3 main folders and go fresh?

We've actually taken great pains to make sure the updater works in Portable Mode. It should be working fine in more recent versions of OBS.

That said, overwriting the files from GitHub zips is definitely not recommended or supported. If using the zip files extracted to a separate location that is not write-protected to run Portable Mode, you should just extract the zip to its own folder and copy/import plugins, profiles, and scene collection data as necessary. You can _probably_ simply delete the non-config directories and then extract the zip contents anew to ensure you aren't ending up with strange files. Generally, however, we expect users setting up Portable Mode from the zip files to extract them to fresh new directories, not overwrite existing ones.


> Just in case, I went ahead and deleted the 3 main folders in the zip, and redownloaded v30.1.0. I expected it to not crash, as it never does after the first time, but I wanted to upload this log, in case something pops up that is different from the earlier ones. [2024-03-23 01-58-04.txt](https://github.com/obsproject/obs-studio/files/14729690/2024-03-23.01-58-04.txt)

When you say, "the first time", what do you mean? The first time after duplicating the profile? The first time after updating OBS? Something else?

**Comment by RaienryuuNoNatsu** [comment]
> Generally, however, we expect users setting up Portable Mode from the zip files to extract them to fresh new directories, not overwrite existing ones.
> 
Yeah, I should do this everytime. I just forget and tend to go a bit lazy mode and that one. Weird, as I still do updates manually, but I'll keep it in mind. Last log is from such fresh unzip at least, so that's one doubt out of the way.

> When you say, "the first time", what do you mean? The first time after duplicating the profile? The first time after updating OBS? Something else?
> 
Oh, I mean when I startup my PC and I open OBS to stream. The very first time I "Start Streaming", that's when it crashes. Any subsequent attempt just works. As such, in-between comments, I tried streaming twice without any crash.

**Comment by RytoEX** [comment]
> > Generally, however, we expect users setting up Portable Mode from the zip files to extract them to fresh new directories, not overwrite existing ones.
> 
> Yeah, I should do this everytime. I just forget and tend to go a bit lazy mode on that one. Weird, as I still do updates manually, but I'll keep it in mind. Last log is from such fresh unzip at least, so that's one doubt out of the way.

The last log doesn't show a crash. It shows a normal OBS session, so it's not much more informative on what is causing the crash.

> > When you say, "the first time", what do you mean? The first time after duplicating the profile? The first time after updating OBS? Something else?
> 
> Oh, I mean when I startup my PC and I open OBS to stream. The very first time I "Start Streaming", that's when it crashes. Any subsequent attempt just works. As such, in-between comments, I tried streaming twice without any crash.

I have no idea why a PC restart would have any bearing on this crash. This just makes less and less sense to me.

**Comment by RaienryuuNoNatsu** [comment]
Yeah, I feel you if this looks weird. This crash also happened on the 20th as well, but I shrugged it off as a random occurance. But then, under this specific case, it happened earlier today. All I can do as a sort of "proof of concept" is to download my backup of 30.0.2, which uses the exact same profiles, and see what happens. It's already 2AM though, so it'll have to be tomorrow after a fresh startup, as I also have a streaming planned. As I go into streaming, I'll post if I crashed or not, and a crash log in case.

**Comment by RaienryuuNoNatsu** [comment]
Good morning. So, here I am again with somewhat of a good news, which makes this even weirder. Before going to sleep yesterday, I downgraded to 30.0.2, and I've just tried starting a streaming after a fresh startup. As expected, no crashes.

After that, I updated again to 30.1.0 and restarted the system. This time, as well, I got no crashes. I can't vouche for it being the solution, as it might strictly be a pure fresh boot, meaningless as it sounds. Nonetheless, it works.

One thing I did was to replace my profiles folder with the one I had backed up from 30.0.2, so maaaaybe, as it was first pointed out, a very slight setting change after the update could've caused this. Before closing this issue, I wanna try again when I can do a full fresh boot on this version.

**Comment by RaienryuuNoNatsu** [comment]
All right. So, as of today, I tried streaming with version 30.1.0 and I got no crashes. I wanna sum it up in case this comes up again for other people, so they don't have to read the whole thing.

So, first and foremost, it was most likely an user error and not a bug in itself. It was pointed out that profiles duplication has some quirks, and there's a referenced issue linked above. I'm gonna go over whatever I did while this issue was open, as a pointer for other people.

The first thing, as I run in portable mode and I update OBS manually by downloading the zips, is to always make a backup of your present installation. As they said, if it ain't broken don't fix it, but it's also true that we wanna try those new features when they come up. So, just keep a working installation backed up, in case something goes wrong.
Secondly, as it was justly explained in here, when updating OBS never overwrite the existing installation. Instead always delete the older folders, and replace them with the new ones from the zip.

With that out of the way, I noticed that the profiles I mostly use for streaming have been modified around the time my first crash occured. I simply downloaded a back up of version 30.0.2, when everything was working just fine, and noticed the profiles' folders had been edited. What broke them is unclear, if not for the fact that it'd only happen once after starting my PC up, after which crash it'd just work out. And while it's true that the function crashing has been pointed out, it's just unclear why that happened.

And with that said, thanks for the support in the matter.
